xc/xenops_server_xen.ml

```diff
@@ -310,6 +310,48 @@ module Storage = struct
   let get_disk_by_name = get_disk_by_name
 end
 
+let print_fork_error f =
+  try
+    f ()
+  with Forkhelpers.Spawn_internal_error(stderr, stdout, status) as e ->
+    begin match status with
+      | Unix.WEXITED n ->
+        error "Forkhelpers.Spawn_internal_error(%s, %s, WEXITED %d)" stderr stdout n;
+        raise e
+      | Unix.WSIGNALED n ->
+        error "Forkhelpers.Spawn_internal_error(%s, %s, WSIGNALED %d)" stderr stdout n;
+        raise e
+      | Unix.WSTOPPED n ->
+        error "Forkhelpers.Spawn_internal_error(%s, %s, WSTOPPED %d)" stderr stdout n;
+        raise e
+    end
+
+let run_command cmd args =
+  debug "running %s %s" cmd (String.concat " " args);
+  let stdout, stderr = print_fork_error (fun () -> Forkhelpers.execute_command_get_output cmd args) in
+  stdout
+
+module NbdClient = struct
+  let start_nbd_client ~unix_socket_path ~export_name =
+    run_command "/opt/xensource/libexec/nbd_client_manager.py" ["connect"; "--path"; unix_socket_path; "--exportname"; export_name]
```

Comment:
Does this one need to be put as a constant somewhere?

Reply:
Also where is this script coming from, I don't see it in this PR?

Reply:
It's in xapi. I've put the constant into `Xapi_globs`, which is only accessible from xapi IMO

Reply:
This is a very unfortunate duplication. Can it be pulled to a commond module? The two things look identical

Reply:
Jon said it's not worth it.

Reply:
However, we could put the `print_fork_error` function into forkexecd at some point, maybe.

Reply:
Could the python script live in this repo though in `scripts/`? It would seem that any bugfixes inthere would be tied with xenopsd changes and then you'd have to open PRs on 2 repos...

Reply:
It's also used in xapi, by the `static-vdis` script, and also in `VBD.plug/unplug`, so I wasn't sure where to put it.

Reply:
If its used by both xapi and xenopsd then lets leave the script where it is now